### PR TITLE
[hotfix] Improve CI trigger and cancel strategy

### PR DESCRIPTION
**Pull request by tisonkun**
Avoid main branch CI jobs being canceled.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,14 +17,33 @@
...
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.event.number || github.run_id }}
   cancel-in-progress: true
+
 jobs:
   compile_and_test:
     uses: apache/flink-connector-shared-utils/.github/workflows/ci.yml@ci_utils
     with:
       flink_version: 1.17.0
       timeout_global: 80
-      timeout_test: 60
```
Comment (syhily) [inline .github/workflows/ci.yml:30]: Why we remove this traveling line end?

Comment (tisonkun) [reply .github/workflows/ci.yml:30]: Isn't here appended a newline?

**Review by reswqa — APPROVED** [review]

**Comment by reswqa** [comment]
Thanks @tisonkun, the explanation of the `Concurrency strategy` in this PR is very useful 👍

**Comment by tisonkun** [comment]
Merging...

It seems the daily CI always failed on compilation now. Let me take a look..

**Pull request merged**